components/list/list-item-generic-layout.js

```diff
@@ -56,18 +57,67 @@ class ListItemGenericLayout extends LitElement {
 		`;
 	}
 
+	connectedCallback() {
+		super.connectedCallback();
+
+		this._preventFocusFromNonActions();
```

Comment:
Your assumption that this is to prevent misuse is correct. It's also for things like the above, where the action for the checkbox is independent from the checkbox itself, thus the checkbox theoretically should not have focusability.

I agree that the DOM walking is expensive and not ideal. I think I forgot about the `focusin` event and assumed that there didn't exist a bubbling event when one of the children of the non `action` slots receives focus/is about to. 🤷 

I will play with this; listening for the `focusin` event from the slot level should work and be infinitely more efficient if I can simply redirect focus to the next expected focusable item.

Reply:
Cool, let me know how it goes.  :)

Note that `focus` event can be captured on the container even though it doesn't bubble.

With the standard capture (if supporting IE11) in `firstUpdated`:
```
.addEventListener('some-event', () => { ... }, true);
```
With standard options (if not supporting IE11)  in `firstUpdated`:
```
.addEventListener('some-event', () => { ... }, { capture: true });
```
With Lit directly in `render` ([ref to doc](https://lit-html.polymer-project.org/guide/writing-templates#add-event-listeners)):
```
constructor() {
    super();
    this._clickHandler = {
        // handleEvent method is required.
        handleEvent(e) { 
            console.log('clicked!');
        },
        // event listener objects can also define zero or more of the event 
        // listener options: capture, passive, and once.
        capture: true,
    }
}
render() {
    return html`<button @click=${this._clickHandler}>Click Me!</button>`;
}
```
^ While we typically always prefer to wire-up directly in the template with the `@`, I think this would be the first time using this approach of the handler **object**.

Reply:
Capturing is totally new to me - thanks for this. That's awesome.

Reply:
I've updated this to use a captured `click` handler and `focus` handler attached to the slot.